pkg/queue/cluster_queue_impl.go

```diff
@@ -221,7 +244,27 @@ func (c *clusterQueueBase) DumpInadmissible() (sets.Set[string], bool) {
 	return elements, true
 }
 
+func (c *clusterQueueBase) Snapshot() []*workload.Info {
+	c.rwm.RLock()
+	defer c.rwm.RUnlock()
+	totalLen := c.heap.Len() + len(c.inadmissibleWorkloads)
+	elements := make([]*workload.Info, 0, totalLen)
+	for _, e := range c.heap.List() {
+		info := e.(*workload.Info)
+		elements = append(elements, info)
+	}
+	for _, e := range c.inadmissibleWorkloads {
+		elements = append(elements, e)
+	}
```

Comment:
could we release the lock at this point?

Reply:
Yes, the sorting can be done outside the lock. Ticketed as a follow up: https://github.com/kubernetes-sigs/kueue/issues/1098

Reply:
Ticketed as: https://github.com/kubernetes-sigs/kueue/issues/1098